examples/api/lib/material/context_menu/editable_text_toolbar_builder.0.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This example demonstrates showing the default buttons, but customizing their
+// appearance.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  MyApp({super.key});
+
+  final TextEditingController _controller = TextEditingController(
+    text: 'Right click or long press to see the menu with custom button.',
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('Custom button appearance'),
+        ),
+        body: Center(
+          child: Column(
+            children: <Widget>[
+              const SizedBox(height: 20.0),
+              TextField(
+                controller: _controller,
+                contextMenuBuilder: (BuildContext context, List<ContextMenuButtonItem> buttonItems, Offset primaryAnchor, [Offset? secondaryAnchor]) {
```

Comment:
I think this goes back to [this discussion above](https://github.com/flutter/flutter/pull/107193#discussion_r933784798).  Do we want to pass simple parameters that are used to build the default menu (list of button items and the anchors), or something comprehensive that gives users more control?

I was leaning towards the "simple" solution of late, but I was assuming that users could look up the EditableTextState if they wanted to.  That's not true anymore though, is it?  Maybe I should just pass BuildContext and EditableTextState, then provide easy ways to get the buttonItems and anchors if needed.

Reply:
I went down a rabbit hole trying to refactor this to pass only EditableTextState ([wip commit](https://github.com/flutter/flutter/commit/b6808d2fa07bb3bd52fc9772d264899915bc74a4)), but I think we should keep the parameters as-is for now, if I can convince you to agree with me.  The two anchors are the simplest representation of the menu position, they existed as part of the toolbar API before this PR, and they also work really well for precise positioning based on right click.  I'd like to make this more customizable for users, but this isn't going to be both clean and comprehensive without a lot of refactoring because we don't have good ways to get/expose metrics about this kind of stuff right now.

I propose we release the simple parameters as-is, and then we see how users react and what they are trying to build with this (maybe they want to hang the menu off the start/end of the selection, maybe they want to point to specific word within the selection, etc.).  Once we know what users want and we know that it's worthwhile to make this more customizable, then we can make that possible in a clean way in a separate PR.

Reply:
I've moved back to passing EditableTextState, details in this comment: https://github.com/flutter/flutter/pull/107193#issuecomment-1268749813